### PR TITLE
Migrate pull-kubernetes-e2e-* jobs to --json

### PR DESCRIPTION
**Pull request by fejta**
/assign @krzyzacy 
Also delete pull-kubernetes-federation-e2e-gce-gci (it no longer matches the non-gci one which is already migrated)

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml [hunk 1]
```diff
@@ -93,42 +93,12 @@
...
         job-name: pull-kubernetes-e2e-kops-aws
         json: 0
         repo-name: 'k8s.io/kubernetes'
         timeout: 0
-    - kubernetes-federation-e2e-gce-gci:
-        max-total: 12
-        job-name: pull-kubernetes-federation-e2e-gce-gci
-        json: 0
-        repo-name: 'k8s.io/kubernetes'
-        timeout: 0
```
Comment (krzyzacy) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml]: deleted?

File: jenkins/bootstrap_test.py [hunk 2]
```diff
@@ -1515,18 +1515,30 @@ def Check(job, name):
...
-                        self.assertEquals(job[attr], job2[attr])
+                        self.assertEquals(job[attr], job2[attr], job)
         for name, job in jobs2.iteritems():
-            job2 = jobs1[name]
+            if job.get('json') and 'kubernetes-e2e' in name:
+                jobs = json_jobs
+                skip_json = True
+            else:
+                jobs = jobs1
+                skip_json =False
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: nit: space

Comment (fejta) [inline jenkins/bootstrap_test.py]: Done

File: jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml (hunk 1 above)
Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml]: Yeah, not used and unclear the best way to migrate, will handle later.

Comment (krzyzacy) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml]: how about the corresponding prow job?

Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml]: Done

**Comment by fejta** [comment]
ref https://github.com/kubernetes/test-infra/issues/1230

**Pull request merged**